### PR TITLE
BLD: Rebased ci improvements 

### PR DESCRIPTION
**Pull request by bashtage**
Rebased and slightly modified version of @matthew-brett 's CI improvements that add pip testing and pre-release testing. The latter is pretty useful and I submitted a PR with fixes for 24 fails that are coming. It also adds pip for appveyor to avoid the issues around 0.9's release. 

I made some slight configuration changes to the travis builds to provide coverage.  Currently

NumPy: 1.11, 1.12, 1.13, 1.14, 1.15
SciPy: 0.18, 0.19, 1, 1.1
Pandas: 0.19, 0.20, 0.22, 0.23 
Matplotlib: 1.5, 2, 2.2

This is at least as good as it was before.   This only required 1 additional job rather than 2 that were in the original PR. 

closes #4652

### CODE REVIEW

File: tools/ci/docbuild_install.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
```
Comment (jbrockmendel) [inline tools/ci/docbuild_install.sh]: Coach is complaining about these she bangs being misssing

**Comment by ChadFulton** [comment]
+1 on improving CI, although sorry to say that I don't understand the details

**Comment by bashtage** [comment]
The main point of this PR is to add testing against pip wheels.  This will avoid release problems where conda packages pass but pip wheels fail.   The other gain from this is testing against pre-releases of NumPy , SciPy and pandas.

File: tools/ci/docbuild_install.sh (hunk 1 above)
Comment (bashtage) [reply tools/ci/docbuild_install.sh]: These are only for use with source and no standalone scripts, so no shebang.

File: statsmodels/regression/tests/test_recursive_ls.py [hunk 2]
```diff
@@ -160,7 +160,7 @@ def test_glm(constraints=None):
 
     # Regression coefficients, standard errors, and estimated scale
     assert_allclose(res.params, res_glm.params)
-    assert_allclose(res.bse, res_glm.bse)
+    assert_allclose(res.bse, res_glm.bse, atol=1e-6)
```
Comment (jbrockmendel) [inline statsmodels/regression/tests/test_recursive_ls.py]: Is this not passing at the moment?

Comment (bashtage) [reply statsmodels/regression/tests/test_recursive_ls.py]: This PR adds new testing environments using PyPI packages which use different compilers and blas.

**Comment by josef-pkt** [comment]
based on the time in the PR testing lists:
change to appveyor is a bit expensive, goes from around 40 min to 1h20min which can make the CI testing queue quite a bit slower and longer when there is more action.

travis goes from just below 3h to around 3h15min, which is not too large an increase (travis runs more simultaneous jobs)

**Comment by matthew-brett** [comment]
For the time of the Appveyor checks - maybe you can drop the conda builds on the basis that conda-forge is testing those for you?    I think y'all do need to test 32-bit Windows, it's significantly different from 64-bit, and still fairly widely used.

**Comment by matthew-brett** [comment]
For a long time the 32 bit installer was the default on Windows.  I am sure
that you can give up on 32 bit in a a couple of years.

**Comment by bashtage** [comment]
I marked a few more tests as slow and added --skip-slow to pytest.  The set of 7 runs finish in about 1 hour now. It woudl probably be best to let the slow tests run on 1 or 2 platforms, probably 1 2.7 and 1 3.6 just to be a bit more careful.

File: appveyor.yml [hunk 3]
```diff
@@ -4,9 +4,18 @@ clone_depth: 50
 os: Visual Studio 2015
 
 environment:
+  PYTEST_DIRECTIVES: --skip-slow
```
Comment (josef-pkt) [inline appveyor.yml]: question: to disable skip-slow temporarily e.g. in a PR, can we just delete `--skip-slow` or does there need to be something there so it's not empty

**Comment by bashtage** [comment]
Empty works.

**Comment by josef-pkt** [comment]
I will merge this with just a surface understanding of the changes.
We will need to turnoff `--skip-slow` in release preparation and maybe some PRs that touch skipped unit tests, 
e.g. We need more work on ZeroInflated countmodels that has now a ZIP marked as slow.
(slowness is largely because we don't have good start_params yet)

**Comment by josef-pkt** [comment]
green enough, merging

@bashtage @matthew-brett  Thank you

**Comment by josef-pkt** [comment]
commit 6b0302fd0f7d298fd22c8948fd3dadfd939726bf closed too many issues, i.e. issues that require the setup.py changes for handling build/install dependencies like #4635

**Pull request merged**